cluster-operator/src/main/java/io/strimzi/operator/cluster/ClusterOperator.java

```diff
@@ -136,11 +136,17 @@ public void start(Promise<Void> start) {
                 }));
             }
 
-            startFutures.add(kafkaConnectAssemblyOperator.createConnectorWatch(namespace, kafkaConnectAssemblyOperator.recreateWatch(namespace)).compose(w -> {
+            startFutures.add(kafkaConnectAssemblyOperator.createKafkaConnectorWatch(namespace, config.getCustomResourceSelector()).compose(w -> {
                 LOGGER.info("Opened KafkaConnector watch for {} operator", kafkaConnectAssemblyOperator.kind());
                 watchByKind.put(KafkaConnector.RESOURCE_KIND, w);
                 return Future.succeededFuture();
             }));
+
+            startFutures.add(kafkaRebalanceAssemblyOperator.createRebalanceWatch(namespace, kafkaRebalanceAssemblyOperator.recreateRebalanceWatch(namespace)).compose(w -> {
```

Comment:
Since the logic of the `createRebalanceWatch` and `recreateRebalanceWatch` of the KafkaRebalanceAssemblyOperator is quite different from the `createWatch` and `recreateWatch` of the AbstractAssemblyOperator, I removed it based on Jakub's feedback.

We could of course move the `KafkaRebalanceAssemblyOperator` watch into the standard `createWatch` loop but it would require us to move the logic of:

- `createRebalanceWatch` -> `createWatch`
- `recreateRebalanceWatch` -> `createWatch`

overriding the inherited `createWatch` and `createWatch` methods from the AbstractAssemblyOperator in the KafkaRebalanceAssemblyOperator class. It doesn't matter to me which path we take, just want everyone to agree on the direction we take

If you feel strongly about it, I'll revist with @scholzj

Reply:
Thanks for the explanation. To me it's just a matter of avoiding code duplication. For example, recreateRebalanceWatch and super.recreateWatch looks exactly the same. Then, createRebalanceWatch looks similar to the standard use case, but not exactly the same, so maybe we could address this in another dedicated PR.

Reply:
I probably don't get what is different here. It should ideally unify and use the `createWatch`. If it needs to be different, then I personally would prefer to not override to make it more visible that it is different.